Node editor2

riteshsamala opened the pull request:
All latest commits of model_ui merged into node_editor 

which also includes handleInitial value (resolves #31), corresponding PM function, type checker module 

also includes prof suggested text changes to quantity editor (resolves #23)

jonwetzel commented on www/index.php:
```diff
@@ -153,41 +153,51 @@
 			
 			<div id="studentModelControl" class="fieldgroup">
-				<label style="width:20ex;" for="setStudentNode">Show to student</label>
+				<label style="width:20ex;" for="setStudentNode">given to student</label>
```
In general all the title text for widgets should start with a capital letter in our dialogs.  Capitalize "G" in "given".

jonwetzel commented on www/index.php:
```diff
@@ -153,41 +153,51 @@
-					<option value='required'>in equation & required</option>
-					<option value='allowed'>in equation & optional</option>
-					<option value='irrelevant'>not in equation</option>
+					<option value='required'>in equations & required</option>
```
Capitalize "In"

jonwetzel commented on www/index.php:
```diff
@@ -153,41 +153,51 @@
-					<option value='allowed'>in equation & optional</option>
-					<option value='irrelevant'>not in equation</option>
+					<option value='required'>in equations & required</option>
+					<option value='allowed'>in equations & optional</option>
```
Capitalize "In"

jonwetzel commented on www/index.php:
```diff
@@ -153,41 +153,51 @@
-					<option value='irrelevant'>not in equation</option>
+					<option value='required'>in equations & required</option>
+					<option value='allowed'>in equations & optional</option>
+					<option value='irrelevant'>not in equations</option>
```
Capitalize "Not"

jonwetzel commented on www/index.php:
```diff
@@ -261,9 +271,16 @@
 			
 			<div id="studentModelControl2" class="fieldgroup">
-				<label style="width:20ex;" for="setStudentNode2">Show to student</label>
+				<label style="width:20ex;" for="setStudentNode2">given to student</label>
```
Capitalize "Given"

riteshsamala commented:
Will check and get back to you
On Fri, Jan 20, 2017 at 2:03 PM Jon Wetzel <notifications@github.com> wrote:

> *@jonwetzel* requested changes on this pull request.
>
>
>
>
> In addition to the text changes in the comments, I have these notes as
> well:
>
>
>
>
>
>    1. Equation Editor changes are missing. (Text area and button
>    movements)
>
>    2. On my desktop the canvas is getting stuck under the menu again;
>    however if I resize the browser window it shifts down and everything looks
>    correct. Can you reproduce this?
>
>
> ------------------------------
>
>
>
>
> In www/index.php
> <https://github.com/Dragoon-Lab/topomath/pull/36#pullrequestreview-17763406>
> :
>
>
> >  		<!-- Putting Node-Editor -Dialog stuff for demo -->
>
>  		<div class="claro sameedit" data-dojo-type="dijit/Dialog" id="quantityNodeEditor">
>
>  			
>
>  			<div id="studentModelControl" class="fieldgroup">
>
> -				<label style="width:20ex;" for="setStudentNode">Show to student</label>
>
> +				<label style="width:20ex;" for="setStudentNode">given to student</label>
>
>
>
> In general all the title text for widgets should start with a capital
> letter in our dialogs. Capitalize "G" in "given".
>
>
>
>
> ------------------------------
>
>
>
>
> In www/index.php
> <https://github.com/Dragoon-Lab/topomath/pull/36#pullrequestreview-17763406>
> :
>
>
> >  				<select id="selectKind" data-dojo-type="dijit/form/Select">
>
>  					<option value='defaultSelect'>--Select--</option>
>
> -					<option value='required'>in equation & required</option>
>
> -					<option value='allowed'>in equation & optional</option>
>
> -					<option value='irrelevant'>not in equation</option>
>
> +					<option value='required'>in equations & required</option>
>
>
>
> Capitalize "In"
>
>
>
>
> ------------------------------
>
>
>
>
> In www/index.php
> <https://github.com/Dragoon-Lab/topomath/pull/36#pullrequestreview-17763406>
> :
>
>
> >  				<select id="selectKind" data-dojo-type="dijit/form/Select">
>
>  					<option value='defaultSelect'>--Select--</option>
>
> -					<option value='required'>in equation & required</option>
>
> -					<option value='allowed'>in equation & optional</option>
>
> -					<option value='irrelevant'>not in equation</option>
>
> +					<option value='required'>in equations & required</option>
>
> +					<option value='allowed'>in equations & optional</option>
>
>
>
> Capitalize "In"
>
>
>
>
> ------------------------------
>
>
>
>
> In www/index.php
> <https://github.com/Dragoon-Lab/topomath/pull/36#pullrequestreview-17763406>
> :
>
>
> >  				<select id="selectKind" data-dojo-type="dijit/form/Select">
>
>  					<option value='defaultSelect'>--Select--</option>
>
> -					<option value='required'>in equation & required</option>
>
> -					<option value='allowed'>in equation & optional</option>
>
> -					<option value='irrelevant'>not in equation</option>
>
> +					<option value='required'>in equations & required</option>
>
> +					<option value='allowed'>in equations & optional</option>
>
> +					<option value='irrelevant'>not in equations</option>
>
>
>
> Capitalize "Not"
>
>
>
>
> ------------------------------
>
>
>
>
> In www/index.php
> <https://github.com/Dragoon-Lab/topomath/pull/36#pullrequestreview-17763406>
> :
>
>
> > @@ -261,9 +271,16 @@
>
>  		<div class="claro sameedit" data-dojo-type="dijit/Dialog" id="equationNodeEditor">
>
>  			
>
>  			<div id="studentModelControl2" class="fieldgroup">
>
> -				<label style="width:20ex;" for="setStudentNode2">Show to student</label>
>
> +				<label style="width:20ex;" for="setStudentNode2">given to student</label>
>
>
>
> Capitalize "Given"
>
>
>
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Dragoon-Lab/topomath/pull/36#pullrequestreview-17763406>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHOTz2DRgnOkinDwE6onY2jjWQbYIc33ks5rUSEfgaJpZM4LosUW>
> .
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>

sachingrover211 commented on www/js/con-author.js:
```diff
@@ -199,6 +223,73 @@ define([
+			var tempInival = tempIni.value.trim();
+			console.log("temporary value is", tempInival);
+			var IniFlag = {status: undefined, value: undefined };
+			if(!((modelType === "given") && (tempInival == '') )){
```
shouldn't we change the given models to authored model as it has been changed in model.js, to keep everything consistent.

riteshsamala replied:
yes , agreed and changed ti "authored" ( as per model)

sachingrover211 commented on www/js/main.js:
```diff
@@ -37,75 +34,117 @@ define([
+                	}
+            	}
+            	_model.active.setPosition(id, index, {"x": g.x, "y": g.y+scrollTop});
+			}, true);
```
I think I skrewed up the indentation here. Would it be possible to open it in sublime and just convert indentation to tabs. 4 indentations to 1 tab space?

riteshsamala replied:
Done

sachingrover211 commented on www/js/typechecker.js:
```diff
@@ -0,0 +1,107 @@
+			});
+		},
+
+		checkInitialValue: function(nodeID, lastInput){
```
Since this function is a generally checks whether you have entered number of not. How about we change the function name to isNumber or something? Typechecker file can be used as a general util file, where we can have such utility functions, and have a singleton object. Or we can externalize validations to this file and use the same idea?

riteshsamala replied:
This is something we have to discuss . Changing the name is quite simple but I think since this also considers the lastInitialValue , it is not just a number checker we might want to keep it as it is.

sachingrover211 replied:
I think, it would be better if we modularize this into two separate functions, in case something comes up in the future. One function can be check numeric value and the other can check last input value. Giving it appropriate names and it starts becoming a general library of small utilities or validation functions.

sachingrover211 replied:
for singleton idea in JS see http://stackoverflow.com/questions/1479319/simplest-cleanest-way-to-implement-singleton-in-javascript

riteshsamala commented:
I tried changing div positions but drawing pane still overlaps a bit into menu bar. I think sachin is the best person to fix this part of UI as he has been working on it @sachingrover211 @jonwetzel . Rest of changes have been made. Please review

sachingrover211 commented:
@riteshsamala I found the issue why this happens. The css for these elements is defined by dojo itself. It is happening like this because, every button in the menu has initially display none attached to it. If we remove the display none then it starts showing with correct css, just like in Dragoon. It was working fine in Dragoon only because the help menu and zoom buttons were always visible in Dragoon. I think to fix this we can use visibility css property. It is like display, but it keeps the dom element in place. Just makes it invisible. Or add some menu element which would be always visible like zoom. The code for it can be directly picked from Dragoon.

riteshsamala commented:
Added visibility property which prevents overlapping of menu bar and following UI

Added two separate functions in typechecker ( checkNumericValue and checkLastInputValue) ... @sachingrover211 Let me know what you think ( I assumed the typechecker as single utility to host both functions and also further similar functions as discussed)

jonwetzel commented on www/js/typechecker.js:
```diff
@@ -94,13 +94,39 @@ define([
+			}
+		},
+
+		checkLastInputValue: function(nodeID, lastInput){
```
I think this should be renamed "checkLastInitialValue" since it is hard-coded to check the initial value field.

jonwetzel commented:
Just waiting on @sachingrover211 to review the new version.

sachingrover211 commented:
@riteshsamala I think these changes are great. Thanks
